solution

JlusS opened the pull request.

JlusS committed:
Create models:
  - User
- Create DAO:
    - UserDAO
- Create service:
    - UserService:
     - AuthenticationService
      - Create your own checked `RegistrationException` and `AuthenticationException`

JlusS committed:
Merge pull request #1

Create models:   - User - Create DAO:     - UserDAO - Create service:     - UserService:      - AuthenticationService       - Create your own checked `RegistrationException` and `AuthenticationException`

mateacademy-ai-mentor commented on src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java:
```diff
@@ -0,0 +1,50 @@
+
+        User user = new User();
+        user.setEmail(email);
+        user.setPassword(password); // Хэш будет сделан в UserService.add
```
Potential issue: You are setting the plain password here and relying on UserService.add to hash it. Please verify that UserService.add actually hashes the password and sets the salt, as required by the checklist. If the checklist requires hashing in AuthenticationServiceImpl, you should hash the password and set the salt here before calling userService.add.

mateacademy-ai-mentor reviewed (changes requested):
### Review Summary

Thank you for your hard work! There is one critical issue to address before approval: your `RegistrationException` is implemented as an unchecked exception (it extends `RuntimeException`), but the requirements specify that both `RegistrationException` and `AuthenticationException` must be checked exceptions (i.e., extend `Exception`). Please update `RegistrationException` to extend `Exception` to fully meet the core requirements. Once this is fixed, your solution will be ready to approve! 😊

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)

> [!IMPORTANT]
> Fix the issues, commit the changes, and then re-request my review using the **`Re-request review` button**[^2] in the Reviewers tab.

[^2]: [Re-request AI review example](https://snipboard.io/nOXxHN.jpg)

> [!TIP]
> If you have any comments about this review, please leave a message in the [Add a comment](#new_comment_field) section below and **mention me with the `@mateacademy-ai-mentor` tag**[^3].
> I will read your message during my next review of your pull request. Please do not reply directly to my comments, as that feature is not supported yet.

[^3]: [Add AI review comments example](https://snipboard.io/hVlOey.jpg)

> [!CAUTION]
> Do not close the pull request until the review is complete. Closing it early can lead to unexpected behavior and disrupt the review process.

JlusS committed:
Create models:
  - User
- Create DAO:
    - UserDAO
- Create service:
    - UserService:
     - AuthenticationService
      - Create your own checked `RegistrationException` and `AuthenticationException`

JlusS committed:
Merge pull request #2

user-service-hw

mateacademy-ai-mentor commented on src/main/java/mate/academy/exception/RegistrationException.java:
```diff
@@ -0,0 +1,7 @@
+package mate.academy.exception;
+
+public class RegistrationException extends RuntimeException {
```
Issue: RegistrationException should extend Exception (checked exception), not RuntimeException. Please update the class to extend Exception as required by the task checklist.

mateacademy-ai-mentor commented on src/main/java/mate/academy/service/impl/UserServiceImpl.java:
```diff
@@ -0,0 +1,27 @@
+    private UserDao userDao;
+
+    @Override
+    public User add(User user) {
+        user.setSalt(HashUtil.getSalt());
+        user.setPassword(HashUtil.hashPassword(user.getPassword(), user.getSalt()));
+        return userDao.add(user);
+    }
```
Issue: Password salting and hashing are performed here, but they are already handled in AuthenticationServiceImpl before calling add. This could result in double hashing/salting, which is not intended. According to the checklist, password hashing and salting should only be done once during registration.

mateacademy-ai-mentor reviewed (approved):
### Review Summary

Great job! 🎉 You have addressed all critical issues, including making RegistrationException a checked exception and ensuring password hashing and salting are only performed once. Your implementation now fully meets the task requirements and checklist. Keep up the good work!

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)
